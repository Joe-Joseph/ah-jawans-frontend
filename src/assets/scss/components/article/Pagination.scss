@import '../../../css/_colors';
@import '../../../css/_radius';

.pagination {
  margin: 10px 10px;
  display: block !important;
  ul li {
    background: none !important;
    button {
      padding: 8px 20px;
      border: none;
      margin-right: 5px;
      cursor: pointer;
      @extend .radius-4;
      outline: 0px dotted;
    }
  }
}
