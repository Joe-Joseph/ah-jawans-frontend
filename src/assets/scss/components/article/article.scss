.articleForm{
    width: 70%;
    margin: auto;

    &__articleInput {
        font-family: 'Poppins', sans-serif;
        font-size: 16px;
        width: 100%;
        margin-bottom:30px;
        border-top: 0;
        border-right: 0;
        border-left: 0;
        outline: none;

        &__add-tag{
            border: 1px solid #cccc;
            width: 80%;
            height: 32px;
            margin-bottom: 30px;
            outline: none;
        }
      }

      &__publish{
        font-family: 'Poppins', sans-serif;
        font-size: 14px;
        width: 120px;
        height: 35px;
        background-color: #12699B;
        color: #F4F0F0;
        border-radius: 5px;
        margin-left:35px
      }
      textarea {
        resize: none;
        border: 0;
        font-family: 'Poppins', sans-serif;
        color: #3f3f3e;
        overflow: auto;
        height: 480px;
    }
}
