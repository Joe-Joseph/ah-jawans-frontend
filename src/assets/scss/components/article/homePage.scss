.header-image {
    width: auto;
    margin: auto;
}

.header-image img{
    position: absolute;
    width: 100%;
    height: 54%;
    top: 80px;
}

.home-page-container{
        margin: auto;
        margin-top: 570px;
    }

.author-article-container{
    width: 80%;
    margin: auto;
    margin-top:410px;
}

.author-articles-container{
    width: 80%;
    margin: auto;
    margin-top:315px;
    background-color: white;
}

.img-card{
    width: 100%;
    height: 120px;
}

.read-time{
    margin-left: 10px;
    color: #C81010;
    font-size: 10px;
}

.author{
    color: #005EA3;
    font-weight: bold;
    font-size: 10px;
}

.article-card {
    height: 240px;
}

.edit-icon img{
    position: absolute;
    width: 20px;
    height: 20px;
    left: 160px; 
}

.remove-icon img{
    position: absolute;
    width: 20px;
    height: 20px;
    left: 200px;
}

@media (max-width: 1440px) {
    .home-page-container {
        margin-top: 510px;
    }
}

@media (max-width: 1024px) {
    .home-page-container {
        margin-top: 450px;
    }
}

@media (max-width: 425px) {
    .header-image img{
        height: 200px;
    }

    .home-page-container {
        margin-top: 250px;
    }
}

@media (max-width: 1047px) {
    .home-page-container {
        margin-top: 450px;
    }
}
