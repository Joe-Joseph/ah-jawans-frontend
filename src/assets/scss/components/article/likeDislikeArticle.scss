.like-button, .dislike-button, .likes-count, .dislikes-count {
    cursor: pointer;
    display: inline-block;
    width: 35px;
    vertical-align: top;
}
.like-button, .dislike-button {
    cursor: pointer;
}
.dislike-button {
    padding-top: 12px;
}
.likes-count, .dislikes-count {
    height: 35px;
    text-align: center;
    font-weight: bold;
    padding-top: 12px;
}
