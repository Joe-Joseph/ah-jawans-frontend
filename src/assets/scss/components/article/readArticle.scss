@import url('https://fonts.googleapis.com/css?family=Poppins&display=swap');
@import url('https://fonts.googleapis.com/css?family=Rum+Raisin&display=swap');

.read-article-body{
    position: relative;
}
.read-article-container{
    margin-top: 130px,
}
.read-article-image{
    width: 75%;
    margin: auto;
    margin-left: 125px;
    max-height:450px
}

.article-text {
    width: 60%;
    margin: auto;
    font-family: 'Poppins', sans-serif;
    color: black;
    text-align: justify;
    margin-top: 50px
}

.title{
    font-family: 'Rum Raisin', sans-serif;
    font-weight: bold;
    color: black;
    text-align: left;
}

.article-title {
    font-family: 'Poppins', sans-serif;
    font-weight: bold;
    font-size: 14px;
    color: black;
}
.author-name {
    padding-left: 700px;
    color: #005EA3;
}

.display-tags {
    position: relative;
    left:27%;
    border: solid 1px #ccc;
    padding: 3px 10px;
    display: inline-block;
    margin-right:10px
}

.display-tags text{
    font-family: 'Poppins', sans-serif;
    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    line-height: 27px;
    color: #8A8888;
    align-items: center;
    text-align: center;
    margin-left:10px
}
.author-image-on-read-article{
   position: relative;
   left: 80px
}
.author-image-on-read-article img{
    position: absolute;
    width: 50px;
    height: 50px;
    left: 140px
 }

 .author-image-on-read-article text{
    position: absolute;
    padding-left: 60px;
    margin-top: 10px;
    left: 140px
 }

 .vote {
    margin-left: 390px;
    margin-top: 30px;
}

@media (max-width: 425px) {
    .read-article-body {
        width: 90%;
    }

    .author-name {
        padding-left: 200px;
    }
}
.comment-form{
    width: 45%;
    margin: 0 auto;
    border: 1px solid #cacdcf;
    padding: 20px;
    margin-top: 50px;
}
.comment-section textarea{
 resize: none;
 font-size: 12px;
 padding: 20px;
}
.comment-btn{
    background-color: #12699b !important;
    font-weight: bold;
    border: none !important;
}

.edit-submit{
    border: none !important;
    background-color: #12699b !important;
}
.form-header img{
    width: 40px;
    border-radius: 50px;
}
.comments{
    margin: 0 auto;
    display: grid;
    grid-gap: 10px;
    grid-template-columns: 1fr;
    margin-top: 20px;
}
.comments .card {
    margin-top: 0px;
}
.comments{
    position: absolute;
}
.authorUsername{
    width: 30px;
    border-radius: 50px;
    margin-right: 20px;
    background: #12699B;
}

.edit-comment-icons{
    position: absolute;
    top:20px;
    left: 50px;
}
.edit-icon-comment button{
   margin-left: 400px;
   background: #ffffff;
}
.remove-icon-comment button{
   margin-left: 420px;
   background: #ffffff;
}
.comments p{
   font-size: 15px;
}
.modal-body textarea{
 resize: none;
 font-size: 14px;
 padding: 20px;
}
.card-header{
    background: #ffffff !important;
}

.comment-edit-icon{
    position: absolute;
    width: 20px;
    height: 20px;
    left: 330px;
    top: 0; 
}

.comment-remove-icon{
    position: absolute;
    width: 20px;
    height: 20px;
    left: 380px;
    top: 0; 
}
