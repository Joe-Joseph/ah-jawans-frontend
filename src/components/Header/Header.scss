@import '../../assets/css/pixels';
@import '../../assets/css/radius';
@import '../../assets/css/colors.scss';
@import '../../assets/css/shadow.scss';
@import '../../assets/css/padding.scss';

.Header {
    top: 0;
    position: fixed;
    @extend .white, .shadow-1;

    .grabSearch {
        margin-top: 10px;
    }

    .logo {
        padding-left: 40px;
    }

    .create-article{
        margin-right: 35px;
    }

    .header-notification-button {
        padding-top: 27px;
        margin-right: 30px;
        .number {
            @extend .radius-5, .danger, .small-text;
            color: $white;
            padding: 2px 5px;
            margin-top: 30px;
            top: 0;
            position: absolute;
        }
    }

    .header-search-button {
        padding-top: 27px;
    }

    .header-create-article-icon img{
        position: absolute;
        width: 20px;
        height: 20px;
        left: 63%;
        top: 45px;
    }

    .header-search-icon img{
        position: absolute;
        width: 20px;
        height: 20px;
        top: 45px;
        right: 21%;
    }

    .create-article-text-header {
        margin-right: 24%;
        color: black;
        outline: none;
        padding-right: 0;
    }
    .create-article-text-header:focus {
        outline: 0px dotted;
    }
    .header-create-article-icon small {
        border: solid 1px red;
    }

    @media screen and (min-width: 761px) {
        .Header {
            .logo {
                padding-left: 40px;
            }

            margin-top: 10px;
            margin-right: 40px;
        }
    }
}

@media screen and (max-width: 760px) {
    .Header {
        .logo {
            padding-left: 15px;
        }

        .header-user-button {
            padding-left: 0px;
            padding-right: 10px;
        }

        .header-search-button {
            padding-left: 0px;
            padding-right: 10px;
        }
    }
}
