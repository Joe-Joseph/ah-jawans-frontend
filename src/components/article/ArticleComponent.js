import React, { Component, Fragment } from 'react';
import CKEditor from '@ckeditor/ckeditor5-react';
import ClassicEditor from '@ckeditor/ckeditor5-build-inline';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import editorConfigs from '../../helpers/ckeditorConfig';
import { articleAction } from '../../redux/actions/articleActions/articleActions';
import { postDataThunkPrivate } from '../../redux/thunks';
import Layout from '../Layout';

export class CreateArticle extends Component {
  state = {
    title: '',
    body: '',
    tags: '',
  }

  componentDidMount() {
    const token = localStorage.getItem('token');
    if (
      !token
      || this.props.article.errors === 'There is no token'
    ) {
      localStorage.clear();
      window.location = '/login';
    }
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    await this.props.postDataThunkPrivate('post', '/articles', articleAction, this.state);
    if (!this.props.article.article.errors) {
      this.props.history.push('/');
      toast.success('article created successfully ');
    } else {
      toast.error(this.props.article.article.errors.title);
      toast.error(this.props.article.article.errors.body);
    }
  }

  render() {
    const { title, body, tags } = this.state;
    return (
      <Layout>
        <Fragment>
          <div className="container-fluid">
            <div className="editor-wrapper">
          <input
              className="form-control article-title"
              value={title}
              placeholder="Title..."
              onChange={(event) => {
                this.setState({ title: event.target.value });
              }}
            />
            <div className="spacer" />
            <div className="body-content">
          <CKEditor
              editor={ClassicEditor}
              data={this.state.body}
              config={editorConfigs}
              onChange={(event, editor) => {
                const data = editor.getData();
                this.setState({ body: data });
              }}
              value={body}
            />
            </div>
            <div className="row">
              <div className="col-md-9">
                <input
                   className="form-control"
                    value={tags}
                    placeholder="Tags..."
                    onChange={(event) => {
                      this.setState({ tags: event.target.value });
                    }}
                />
              </div>
            <div className="col-md-3">
              <button
              className="btn btn-primary art-btn"
              type="button"
                onClick={this.handleSubmit}>
                Publish
              </button>
              </div>
              </div>
          </div>
          </div>
        </Fragment>
      </Layout>
    );
  }
}

CreateArticle.propTypes = {
  article: PropTypes.object.isRequired,
  postDataThunkPrivate: PropTypes.func,
  history: PropTypes.object,
  errors: PropTypes.object,
};

const actionCreator = {
  postDataThunkPrivate,
};

const mapStateToProps = (state) => ({
  article: state.article,
  errors: state.article.article.errors,
});

export default connect(
  mapStateToProps,
  actionCreator,
)(CreateArticle);
