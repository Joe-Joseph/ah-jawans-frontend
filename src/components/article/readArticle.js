/* eslint-disable no-irregular-whitespace */
/* eslint-disable no-unused-expressions */
/* eslint-disable class-methods-use-this */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toast, ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import ReactHtmlParser from 'react-html-parser';
import moment from 'moment';
import ReactImageFallback from 'react-image-fallback';
import Layout from '../Layout/Layout';
import { getDataThunk, postDataThunkPrivate } from '../../redux/thunks/index';
import {
  likeArticleAction,
  dislikeArticleAction,
  getArticleLikesAction,
  getArticleDislikesAction,
  clearLikesOrDislikes,
} from '../../redux/actions/likeDislikeAction';
import {
  createCommentAction, getAllCommentsAction, deleteCommentAction, updateCommentAction,
} from '../../redux/actions/commentsActions/commentsActions';
import getArticlesAction from '../../redux/actions/getArticlesAction';
import { postbookmarkAction } from '../../redux/actions/user/BookmarkAction';
import '../../assets/scss/components/article/readArticle.scss';
import '../../assets/scss/components/article/sharingArticle.scss';
import facebookIcon from '../../assets/images/facebook-logo.png';
import gmail from '../../assets/images/share-on-email.jpg';
import twitter from '../../assets/images/twitter-logo.png';
import LikeDislike from './likeDislikeArticle';
import Bookmark from '../../assets/images/Bookmark.png';
import { CommentComponent } from '../comments/CommentComponent';
import remove from '../../assets/icons/delete.png';
import edit from '../../assets/icons/edit.png';
import profileImagePlaceHolder from '../../assets/images/profile_plaholder.png';
import LoadingGif from '../../assets/images/loadingGif.gif';
import Tags from './displayTags';

export class ReadArticle extends Component {
  state = {
    username: '',
    title: '',
    body: '',
    image: null,
    likes: 0,
    dislikes: 0,
    readers: 0,
    bookmark: '',
    errors: '',
    modal: 'none',
    comment: {
      body: '',
    },
  }

  handleClick = async () => {
    let articleId;
    if (localStorage.getItem('id') !== null) {
      articleId = localStorage.getItem('articleId');
      await this.props.postDataThunkPrivate('post', `bookmarks/${articleId}`, postbookmarkAction);
      this.setState({ errorsbookmark: this.props.errors, bookmark: this.props.bookmark });
      const { errorsbookmark, bookmark } = this.props;
      if (bookmark) { (toast.success(`${bookmark}!`), setTimeout(() => window.location.reload(), 6000)); } else {
        (toast.error(`${errorsbookmark}!`), setTimeout(() => window.location.reload(), 6000));
      }
    } else {
      toast.error('Please Login first to bookmark the article!');
    }
  };

  componentDidMount = async () => {
    const { id } = this.props.match.params;
    await this.props.getDataThunk('get', `articles/${id}`, getArticlesAction);
    const { title, body, readers } = this.props.articles;
    localStorage.setItem('articleId', this.props.articles.id);
    const { username, image } = this.props.articles.author;
    this.setState({
      username,
      title,
      body,
      image,
      readers,
    });
    await this.props.getDataThunk('get', `articles/${id}/likes`, getArticleLikesAction);
    await this.props.getDataThunk('get', `articles/${id}/dislikes`, getArticleDislikesAction);
    this.setState({ username: this.props.articles.author.username });
    await this.props.postDataThunkPrivate('get', `articles/${id}/comments`, getAllCommentsAction, null);
    await this.props.getDataThunk('get', `articles/${id}/likes`, getArticleLikesAction);
    await this.props.getDataThunk('get', `articles/${id}/dislikes`, getArticleDislikesAction);
    this.setState({
      username: this.props.articles.author.username,
    });
  }

  componentWillUnmount = () => {
    this.props.clearLikesOrDislikes();
  }

  onChange = (event) => {
    const comment = { ...this.state.comment };
    comment[event.target.name] = event.target.value;
    this.setState({ comment });
  }

  handleDelete = (commentId) => async () => {
    const { id } = this.props.match.params;
    await this.props.postDataThunkPrivate('delete', `articles/${id}/comments/${commentId}`, deleteCommentAction, null);
    window.location.reload();
  }

  onComment = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleUpdate = (commentId) => async () => {
    const { id } = this.props.match.params;
    const { body } = this.state;
    await this.props.postDataThunkPrivate('patch', `articles/${id}/comments/${commentId}`, updateCommentAction, { body });
    this.props.comments.errors ? (
      toast.error(this.props.comments.errors.body))
      : (window.location.replace(`/readArticle/${id}`),
      toast.success('comment updated.'));
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { id } = this.props.match.params;
    const { comment } = this.state;
    await this.props.postDataThunkPrivate('post', `articles/${id}/comments`, createCommentAction, comment);
    this.props.comments.errors ? (
      toast.error(this.props.comments.errors.body))
      : (setTimeout(() => window.location.reload(), 6000),
      (toast.success('comment created.'), setTimeout(() => window.location.reload(), 6000)));
  }

  hideModal = () => {
    this.setState({ modal: 'none' });
  }

  handleLike = async () => {
    const { id } = this.props.match.params;
    if (localStorage.getItem('id') === null) {
      toast.error('Login first to like or dislike this article!');
    } else { await this.props.postDataThunkPrivate('post', `articles/${id}/like`, likeArticleAction, null); }
    await this.props.getDataThunk('get', `articles/${id}/likes`, getArticleLikesAction);
    await this.props.getDataThunk('get', `articles/${id}/dislikes`, getArticleDislikesAction);
  }

componentWillReceiveProps = (nextProps) => {
  const { dislikes, likes } = nextProps;
  this.setState((prevState) => ({
    ...prevState,
    likes: likes || prevState.likes,
    dislikes: dislikes || prevState.dislikes,
  }));
}

handleDislike = async () => {
  const { id } = this.props.match.params;
  if (localStorage.getItem('id') === null) {
    toast.error('Login first to like or dislike this article!');
  } else { await this.props.postDataThunkPrivate('post', `articles/${id}/dislike`, dislikeArticleAction, null); }
  await this.props.getDataThunk('get', `articles/${id}/dislikes`, getArticleDislikesAction);
  await this.props.getDataThunk('get', `articles/${id}/likes`, getArticleLikesAction);
}

componentWillUnmount = () => {
  this.props.clearLikesOrDislikes();
}

render() {
  const { likes, dislikes } = this.props;
  const {
    title, body, readers, username, image,
  } = this.state;
  const { id } = this.props.match.params;
  const url = `https://ah-jawans-frontend.herokuapp.com/readArticle/${id}`;
  const tweet = `'${this.state.title}' -by ${this.state.username} @ https://ah-jawans-frontend.herokuapp.com/readArticle/${id}`;

  return (
      <Layout>
        <div className='read-article-body'>
         <ToastContainer position={toast.POSITION.TOP_RIGHT} />
          <div className='container read-article-container'>
            <h2 className='article-text title'>{title}</h2>
            <br />
            { username && <div className='author-image-on-read-article author-name'>
            <ReactImageFallback
              src={image && (image.split(':')[0] === 'https') ? image : `https://res.cloudinary.com/djxhcwowp/image/upload/v${image}`}
              fallbackImage={profileImagePlaceHolder}
              initialImage={LoadingGif}
              alt="profile image"
              className="authorUsername"
            />
              <text>{username}</text>
              </div>}
              <br/>
            <p className='article-text'>{ReactHtmlParser(body)}</p>
            <br />
          </div>
          {body && <div className='sharing-icons'>
            <a href={`https://www.facebook.com/sharer/sharer.php?u=${url}`}>
              <img className="" src={facebookIcon} alt="facebook" />
            </a>
            <a href={`https://twitter.com/intent/tweet?text=${tweet}`}>
              <img className="" src={twitter} alt="twitter" />
            </a>
            <a href={`mailto:?subject=${title}&body="${title}" by ${username} @ ${url}`}>
              <img className="" src={gmail} alt="gmail" />
            </a>
            <a
              id="bookmarkarticle"
              onClick={this.handleClick}
            >
              <img className="" src={Bookmark} alt="Bookmark" />
            </a>
          </div>}
          {this.props.articles.tagList && this.props.articles.tagList.map(
            (tag, index) => <Tags key={index} tagText={tag}/>,
          )}

          { body && <div className='vote'><LikeDislike
              handleLike={this.handleLike}
              handleDislike={this.handleDislike}
              likes={likes}
          dislikes={dislikes} /><span className="views">{`Views:Â ${readers}`}</span></div>}

        <div id="comment-section" className="comment-section">
          {(localStorage.username)
            && <CommentComponent
              onChange={this.onChange}
              onSubmit={this.handleSubmit}
              comment={this.state.comment}
            />
          }
          </div>
          < div className='col-md-5 comments'>
            {this.props.comments.allComments
              && this.props.comments.allComments.map((comment, index) => (
                <div className="card" key={comment.id}>
                  <div className="card-header">
                    <ReactImageFallback
                      src={comment.User.image && (comment.User.image.split(':')[0] === 'https') ? comment.User.image : `https://res.cloudinary.com/djxhcwowp/image/upload/v${comment.User.image}`}
                      fallbackImage={profileImagePlaceHolder}
                      initialImage={LoadingGif}
                      alt="profile image"
                      className="authorUsername"
                    />
                  </div>
                  <div className='edit-comment-icons'>
                    <img
                      src={edit} onClick={() => this.setState({
                        modal: 'block',
                        body: comment.body,
                        id: comment.id,
                      })}
                      id={`comment-edit-icon${index}`}
                      className="comment-edit-icon"
                    />
                      <img
                        src={remove}
                        id={`comment-remove-icon${index}`}
                        className="comment-remove-icon"
                        onClick={this.handleDelete(comment.id)}
                      />
                  </div>
                  <div className="card-body">
                    <blockquote className="blockquote mb-0">
                      <p>{comment.body}</p>
                      <small className="text-muted">{moment(comment.updatedAt).fromNow()}</small>
                    </blockquote>
                  </div>
                  <div className={`modals ${this.state.modal}`}>
                    <div className="modal-dialog modal-dialog-centered">
                      <div className="modal-content">
                        <div className="modal-header">
                          <h5 className="modal-title">Edit comment</h5>
                          <button id={`hideModel${index}`} type="button" className="close" onClick={this.hideModal}>
                            <span aria-hidden="true" >&times;</span>
                          </button>
                        </div>
                        <div className="modal-body">
                          <textarea id={`textarea${index}`} className="form-control comment-body" name="body" value={this.state.body} onChange={this.onComment.bind(this)} />
                        </div>
                        <div className="modal-footer">
                          <button type="button" className="btn btn-secondary" data-dismiss="modal" onClick={this.hideModal}>Cancel</button>
                          <button id={`updatebutton${index}`} type="button" className="btn btn-primary edit-submit" onClick={this.handleUpdate(this.state.id)}>Save changes</button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              ))}
          </div>
        </div>
      </Layout>
  );
}
}

ReadArticle.propTypes = {
  getDataThunk: PropTypes.func,
  postDataThunkPrivate: PropTypes.func,
  articles: PropTypes.object,
  match: PropTypes.object,
  author: PropTypes.string,
  likes: PropTypes.number,
  dislikes: PropTypes.number,
  clearLikesOrDislikes: PropTypes.func,
  bookmark: PropTypes.string,
  errorsbookmark: PropTypes.string,
  location: PropTypes.object,
  comments: PropTypes.object,
  history: PropTypes.object,
  errors: PropTypes.object,
  id: PropTypes.string,
  comment: PropTypes.object,
};

const mapStateToProps = (state) => ({
  articles: state.articles.articles,
  likes: state.likeDislikeReducer.likes.count,
  dislikes: state.likeDislikeReducer.dislikes.count,
  bookmark: state.bookmark.message,
  errorsbookmark: state.bookmark.errors,
  comments: state.comments.comments,
  errors: state.comments.errors,
  userProfile: state.getUser,
});

const actionCreator = {
  getDataThunk,
  postDataThunkPrivate,
  clearLikesOrDislikes,
};

export default connect(mapStateToProps, actionCreator)(ReadArticle);
