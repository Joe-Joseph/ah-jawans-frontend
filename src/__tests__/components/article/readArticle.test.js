import React from 'react';
import { Provider } from 'react-redux';
import { mount, shallow } from '../../../../config/enzymeConfig';
import { MemoryRouter } from 'react-router-dom';
import { ReadArticle as ReadArticleComponent } from '../../../components/article/readArticle';
import configureStore from '../../../redux/store/index';
import '../../../components/Header/Header';

jest.mock('../../../components/Header/Header', () => () => (
  <div id="mocked">hello</div>
))

const store = configureStore();

const props = {
  articles: {},
  postDataThunkPrivate: jest.fn(),
  getDataThunk: jest.fn(),
  likes: 0,
  dislikes: 0,
  articles: {
    id: 1,
    username: 'Joe',
    title: 'This is title of the article',
    body: 'This is the body of the article',
    author: { username: 'shaluc' }
  },
  bookmark: { message: {} },
  errors: { errors: {} },
  match: { params: 3 },
  handleClick: jest.fn(),
}

const state = {
  id: 1,
  username: 'Joe',
  title: 'This is title of the article',
  body: 'This is the body of the article',
  author: { username: 'shaluc' },
  bookmark: { message: {} },
  errors: { errors: {} },
  postDataThunkPrivate: jest.fn(),
  getDataThunk: jest.fn(),
  match: { params: 3 },
  handleClick: jest.fn(),
  likes: 0,
  dislikes: 0,
}

describe('<ReadArticle/>', () => {
  it('Should render', () => {
    localStorage.setItem('id', ' ');
    const wrapper = mount(
      <MemoryRouter>
        <ReadArticleComponent {...props} />
      </MemoryRouter>
    );
    expect(wrapper.contains(<div></div>));
  });

  it('Should render', () => {
    const wrapper = shallow(
      <ReadArticleComponent {...props} />
    );
    wrapper.setProps(props);
    wrapper.instance().handleLike();
    wrapper.instance().handleDislike();
    expect(wrapper.contains(<div></div>));
  });

  it('Should render', () => {
    localStorage.setItem('id', '111');
    const wrapper = shallow(
      <ReadArticleComponent {...props} />
    );
    wrapper.setProps(props);
    wrapper.instance().handleLike();
    wrapper.instance().handleDislike();
    expect(wrapper.contains(<div></div>));
  });

  const component = shallow(<ReadArticleComponent {...state} />)
  it('should create <View /> snapshot ', () => {
    expect(component).toMatchSnapshot();
  });
  it('should check bookmark ', () => {
    component.instance().handleClick(49);
  });
  it('should bookmark article ', () => {
    component.setProps({ errors: 'eerrorr', bookmark: 'heypassed', body: 'hey body' });
    component.setState({ username: 'shalu', title: 'hey dear', body: 'hey body' });
    localStorage.setItem('id', '9');
    localStorage.setItem('articleId', '19');
    component
      .find('#bookmarkarticle')
      .at(0)
      .simulate('click');
    component.instance().handleClick({ errors: 'eerrorr', bookmark: 'heypassed', body: 'hey body' });
  });
});
