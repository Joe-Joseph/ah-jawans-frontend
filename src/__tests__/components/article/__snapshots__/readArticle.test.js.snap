// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadArticle/> on should create <View /> snapshot  1`] = `ShallowWrapper {}`;
